solidity/contracts/StakingPortBacker.sol

```diff
@@ -0,0 +1,273 @@
+/// all tokens in the new staking contract would be blocked until the
+/// temporarily-staked tokens provided by the Keep team are repaid by the token
+/// owner.
+///
+/// The expected mode of usage for users who have already staked on the old
+/// contract is to:
+/// 1. Copy stake.
+/// 2. Undelegate from the old staking contract.
+/// 3. Operate as normal on the new staking contract.
+/// 4. Once the old staking contract's undelegated balance is available for
+/// recovery, recover it and use the recovered tokens to repay the
+/// StakingPortBacker contract.
+contract StakingPortBacker is Ownable {
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+    using BytesLib for bytes;
+    using RolesLookup for address payable;
+
+    event StakeCopied(
+        address indexed owner,
+        address indexed operator,
+        uint256 value
+    );
+    event StakePaidBack(
+        address indexed owner,
+        address indexed operator
+    );
+    event TokensWithdrawn(uint256 amount);
+
+    /// @notice The maximum allowed time for the token owner to repay the
+    /// delegation backed by tokens from this contract. After this time, the
+    /// owner of the contract can undelegate and recover not paid back
+    /// delegations.
+    uint256 public constant maxAllowedBackingDuration = 7776000; // ~3 months
+
+    IERC20 public keepToken;
+    TokenGrant internal tokenGrant;
+    StakeDelegatable public oldStakingContract;
+    TokenStaking public newStakingContract;
+
+    struct CopiedStake {
+        address owner;
+        uint256 amount;
+        uint256 timestamp;
+        bool paidBack;
+    }
+
+    mapping(address => bool) public allowedOperators; // operator -> allowed to copy?
+    mapping(address => CopiedStake) public copiedStakes; // operator -> copied stake info
+
+    constructor(
+        IERC20 _keepToken,
+        TokenGrant _tokenGrant,
+        StakeDelegatable _oldStakingContract,
+        TokenStaking _newStakingContract
+    ) public {
+        keepToken = _keepToken;
+        tokenGrant = _tokenGrant;
+        oldStakingContract = _oldStakingContract;
+        newStakingContract = _newStakingContract;
+    }
+
+    /// @notice Lets the owner of the contract to register operator from staking
+    /// relaitonships that will be allowed to use the token supply provided by
+    /// this contract.
+    /// @dev The reason for explicitly listing which relationships are allowed
+    /// to use the supply is to avoid a situation when someone stakes on the old
+    /// contract later, after deploying the new staking contract version, just
+    /// to drain the supply from this contract.
+    /// @param operator The operator from the staking relationship on the old
+    /// staking contract that will be allowed to use the token supply provided
+    /// by this contract.
+    function allowOperator(address operator) public onlyOwner {
```

Comment:
Any chance we can make this take an array?

Reply:
Yeah... that's a good point. I think I'll add another function `allowOperators` with an array calling this one in a loop.